main/src/ca/uwaterloo/flix/language/phase/unification/BddFormula.scala

```diff
@@ -15,10 +15,166 @@
  */
 package ca.uwaterloo.flix.language.phase.unification
 
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Symbol, Type}
+import ca.uwaterloo.flix.util.InternalCompilerException
+import ca.uwaterloo.flix.util.collection.Bimap
+
+import scala.collection.immutable.SortedSet
+import org.sosy_lab.pjbdd.api.{Builders, DD}
+
 object BddFormula {
 
-  class Bdd
+  val creator = Builders.bddBuilder().build()
+
+  class BddFormula(val dd: DD) extends Formula { }
```

Comment:
This is just a wrapper around `DD`, but I think you can actually skip it and implement `BoolAlg` directly on `DD`. Does that work/make sense?

Reply:
I need a supertype for BoolFormulas and BddFormulas for HybridFormula to work the way it does right now. But if you want me to change it as described here: https://github.com/flix/flix/pull/5018#discussion_r1035854300, then it should work with BoolAlg[DD]